### ISSUE TITLE
Failed constraint checks and partial indexes should raise E_VALIDATION

### ISSUE THREAD

**Issue by kevinburkeshyp** [balderdashy/sails-postgresql#186]
say i have a table like this

```sql
CREATE TABLE users (
    "account" integer CHECK "account" >= 0
);
```

and I try to write `-1` as the account balance - this will raise a check failure! in theory this should be a 400 E_VALIDATION instead of a 500 server error, but the latter gets raised. 

Similar thing happens for partial indexes - the code matches but they don't match the regex at the bottom of `adapter.js`.

```
CREATE UNIQUE INDEX unique_status
    ON accounts ("driverId")
    WHERE "processingState" = 'ACTIVE';
```

which says, the same driverId can't be set on two different accounts that have the processingState ACTIVE. dumb example sorry but the ones we use this for are currently proprietary.

**Comment by particlebanana** [comment]
We don't have that fine grained error code checking. It would be great to add it if anyone has the time. Looks like the list of possible error codes is [here](http://www.postgresql.org/docs/9.4/static/errcodes-appendix.html).

**Comment by sailsbot** [comment]
Thanks for posting, @kevinburkeshyp.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!

**Comment by devinivy** [comment]
What error does the adapter provide?  Waterline doesn't know anything about HTTP errors (400 vs 500, etc.).

**Comment by sailsbot** [comment]
Thanks for posting, @kevinburkeshyp.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!

**Comment by particlebanana** [comment]
Added this to the roadmap as a feature. https://github.com/balderdashy/sails-postgresql/commit/830f50e8baa9367a8e55ea01c0cf52dde4dbc06b

**Comment by sailsbot** [comment]
Thanks for posting, @kevinburkeshyp.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
